File: TDS_Client/Window/mainNew/src/app/components/mapcreator/models/mapCreatorPosition.ts
import { MapCreatorPositionType } from '../enums/mapcreatorpositiontype.enum';
export class MapCreatorPosition {
    public Id: number;
    public Type: MapCreatorPositionType;
    public Info?: string | number;
    public PosX: number;
    public PosY: number;
    public PosZ: number;
    public RotX: number;
    public RotY: number;
    public RotZ: number;

    constructor(id: number, type: MapCreatorPositionType, posX: number, posY: number, posZ: number, rotX: number, rotY: number, rotZ: number) {
        this.Id = id;
        this.Type = type;
        this.PosX = posX;
        this.PosY = posY;
        this.PosZ = posZ;
        this.RotX = rotX;
        this.RotY = rotY;
        this.RotZ = rotZ;
    }
}

File: TDS_Common/Dto/Map/Creator/MapCreatorPosition.cs
using MessagePack;
using TDS_Common.Enum;

namespace TDS_Common.Dto.Map.Creator
{
    [MessagePackObject]
    public class MapCreatorPosition
    {
        [Key(0)]
        public int Id { get; set; }
        [Key(1)]
        public EMapCreatorPositionType Type { get; set; }
        [Key(2)]
        public object Info { get; set; }
        [Key(3)]
        public float PosX { get; set; }
        [Key(4)]
        public float PosY { get; set; }
        [Key(5)]
        public float PosZ { get; set; }
        [Key(6)]
        public float RotX { get; set; }
        [Key(7)]
        public float RotY { get; set; }
        [Key(8)]
        public float RotZ { get; set; }

        public MapCreatorPosition() { }
    }
}
